lib/URI.pm

```diff
@@ -1085,6 +1116,34 @@ examples:
 This value can be set to ";" to have the query form C<key=value> pairs
 delimited by ";" instead of "&" which is the default.
 
+=back
+
+=head1 ENVIRONMENT VARIABLES
+
+=item URI_HAS_RESERVED_SQUARE_BRACKETS
+
+Before version 5.11, URI treated square brackets as reserved characters
+throughout the whole URI string. However, these brackets are reserved
+only within the autority/host part of the URI and nowhere else (RFC 3986).
+
+Starting with version 5.11, URI takes this distinction into account.
+Setting the environment variable c<URI_HAS_RESERVED_SQUARE_BRACKETS>
+(programmatically or via the shell), restores the old behavior.
+
+ Note: This environment variable is just an initialiser and has to be set
+       before module URI is used/required. Changing it at run time has no effect.
+
+ #-- restore 5.10 behavior programmatically
+ BEGIN {
+   $ENV{URI_HAS_RESERVED_SQUARE_BRACKETS} = 1;
+ }
+ use URI ();
```

Comment:
Ideally, this bit should be indented two spaces so it's highlighted as code in POD and consistent with the rest of the file.

Reply:
I'll fix that and probably will also add a simpler alternative suggestion using a specific version of URI.